### DISCUSSION TITLE
Signed distance map or Distance transform map in MONAI

### DISCUSSION THREAD

**Discussion by YingdiZhang** [Project-MONAI/MONAI#4603] [Q&A]
Thanks for the great job.
I would like to use signed distance map or distance transform map in my 3D 
 medical image segmentation task.  I wonder if there is an implementation in MONAI to transform a 3D mask to signed distance map?

**Comment by tvercaut** [comment]
@masadcv just release [FastGeodis](https://github.com/masadcv/FastGeodis), a package to compute Geodesic and Euclidean distance maps in PyTorch. It does not (yet?) feature signed distance maps but you can always compute two distance maps (one with the mask and another one with the inverted mask) and substract these.

There is also a related issue in MONAI: #1332

And now a feature request in FastGeodis: https://github.com/masadcv/FastGeodis/issues/9

**Reply by masadcv** [reply]
This has now been added to FastGeodis

Here is a simple example usage of signed distance transform:
https://github.com/masadcv/FastGeodis/blob/master/samples/simpledemo2d_signed.ipynb

**Comment by grlee77** [comment]
We plan to have an equivalent of SciPy's `distance_transform_edt` (Euclidean distance transform) for CuPy arrays in the upcoming cuCIM 22.08 release. It will support 2D or 3D data.

in progress PR: https://github.com/rapidsai/cucim/pull/318

We will also look at the signed distance transform as we want to provide a GPU equivalent to ITK's [SignedMaurerDistanceMapImageFilter](https://itk.org/Doxygen/html/classitk_1_1SignedMaurerDistanceMapImageFilter.html). The plan is to compute this by subtracting the output of two calls as discussed above.
